### PR TITLE
Allow System Config of Type Serialized Array to <depend> on other field 

### PR DESCRIPTION
**Pull request by pquerner**
### Description (*)

If you have a serialized array config and want it to "\<depend\>" on some… other field, in the original core this wasn't possible. The js simply didn't work in this scenario and thus the field was always visible.

This branch fixes this, by allowing both: regular (normal fields) and "serialized array" type of fields.

### Related Pull Requests
None

### Fixed Issues (if relevant)
None

### Manual testing scenarios (*)

1. Find a system config which is type "serialized array" (ie: design/theme/template_ua_regexp)
2. Add this field right above this config temporarly 

```xml
<xx translate="label">
    <label>xx</label>
    <frontend_type>select</frontend_type>
    <source_model>adminhtml/system_config_source_yesno</source_model>
    <sort_order>10</sort_order>
    <show_in_default>1</show_in_default>
    <show_in_website>1</show_in_website>
    <show_in_store>1</show_in_store>
</xx>
```

3. Add a "\<depends\>" node to its declaration (app/code/core/Mage/Core/etc/system.xml around line 339)

```xml
<depends><xx><value>1</value></xx></depends>
```

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All automated tests passed successfully (all builds are green)
 - [ ] Add yourself to contributors list

### CODE REVIEW

File: js/mage/adminhtml/form.js [hunk 1]
```diff
@@ -536,22 +550,22 @@ FormElementDependenceController.prototype = {
...
-            $(idTo).up(this._config.levels_up).select('input', 'select', 'td').each(function (item) {
+            ele.select('input', 'select', 'td').each(function (item) {
                 // don't touch hidden inputs (and Use Default inputs too), bc they may have custom logic
                 if ((!item.type || item.type != 'hidden') && !($(item.id+'_inherit') && $(item.id+'_inherit').checked)
                     && !(currentConfig.can_edit_price != undefined && !currentConfig.can_edit_price)) {
                     item.disabled = false;
                 }
             });
-            $(idTo).up(this._config.levels_up).show();
+            this.showElem(ele);
```
Comment (kiatng) [inline js/mage/adminhtml/form.js:560]: I feel that `ele.show()` and later `ele.hide()` is cleaner. What do you think?

**Comment by kiatng** [comment]
Tested and worked. I tested like this:

https://github.com/OpenMage/magento-lts/blob/8681773d0e1d6f5c4e2395bbdd9c88f5ddcc8858/app/code/core/Mage/Core/etc/system.xml#L329-L333

After line 333, add
```xml
<depends><template><value>abc</value></template></depends>
```
When template is "abc":
image
When template is not:
image

File: js/mage/adminhtml/form.js [hunk 2]
```diff
@@ -507,8 +512,17 @@ FormElementDependenceController.prototype = {
...
     {
+        let upLevels = this._config.levels_up;
+        let ele;
         if (!$(idTo)) {
-            return;
+            if ($('row_' + idTo)) {
+                idTo = 'row_' + idTo;
+                ele = $(idTo);
+            } else {
+                return;
```
Comment (kiatng) [inline js/mage/adminhtml/form.js:522]:
```suggestion
            idTo = 'row_' + idTo;
            ele = $(idTo);
            if (!ele) {
                return;
```

Comment (pquerner) [reply js/mage/adminhtml/form.js:522]: Maybe we should put the 

```
if (!ele) {
    return;
}
```

beneath all, when we sorted which "element" we want to try to attach to.

What do you think?

Comment (kiatng) [reply js/mage/adminhtml/form.js:522]: sure.

Comment (pquerner) [reply js/mage/adminhtml/form.js]: After consideration I just accepted your change-request as-is.

**Review by kiatng — APPROVED** [review]

**Review by fballiano — APPROVED** [review]

**Pull request merged**